### PR TITLE
Install iproute2

### PR DESCRIPTION
**Pull request by joestringer**
This allows certain tests like the sk_assign selftests to run.

I tried out these commands via a docker container something like the following, but I don't know how to actually use the mkrootfs here without some arch-like environment (it fails in the docker container run like below):

```
$ docker run --tty --interactive --rm archlinux bash
```

Related: #144

### CODE REVIEW

**Comment by wat-ze-hex** [comment]
Tried several `mkrootfs.sh` modifications: one is based on this PR and a few of my own.
They are all failing with mounting issue I don't understand yet, e.g.:
```
+ cat /etc/fstab
dev /dev devtmpfs rw,nosuid 0 0
proc /proc proc rw,nosuid,nodev,noexec 0 0
sys /sys sysfs rw,nosuid,nodev,noexec 0 0
debugfs /sys/kernel/debug debugfs mode=755,realtime 0 0
bpffs /sys/fs/bpf bpf realtime 0 0
+ /bin/mount -av --fstab /etc/fstab
+ ls -l /proc/mounts
ls: cannot access '/proc/mounts': No such file or directory
+ true
+ mount
mount: failed to read mtab: No such file or directory
```

Any ideas what can be wrong?

File: travis-ci/vmtest/mkrootfs.sh [hunk 1]
```diff
@@ -87,7 +87,34 @@ packages=(
...
+
+pacman -Sy --noconfirm "${makedeps[@]}"
+asp export iproute2
+chmod 777 iproute2
+
+cd iproute2
+sed -i 's/'"'"'iptables'"'"' //g' PKGBUILD
+sed -i '/makedepends/d' PKGBUILD
+chmod 666 *
+su nobody -s /bin/bash -c "makepkg -c --skippgpcheck"
```
Comment (wat-ze-hex) [inline travis-ci/vmtest/mkrootfs.sh:111]: `makepkg` expects some env vars to be set, e.g. `BUILDDIR`, `PKGDEST`, `SRCDEST`.

File: travis-ci/vmtest/mkrootfs.sh [hunk 2]
```diff
@@ -87,7 +87,34 @@ packages=(
...
+	fakeroot
+        flex
+	gcc
+        gettext
+	make
+	patch
+	pkgconf
+)
+
+pacman -Sy --noconfirm "${makedeps[@]}"
```
Comment (wat-ze-hex) [inline travis-ci/vmtest/mkrootfs.sh:103]: Shouldn't `pacman` run in the `chroot` environment passed w/ `-r` option, e.g [0] ?
Without `-r` we're altering the build system, not the target fs.

`arch-chroot $root` should also work.

[0] https://git.archlinux.org/arch-install-scripts.git/tree/pacstrap.in#n115

File: travis-ci/vmtest/mkrootfs.sh [hunk 3]
```diff
@@ -87,7 +87,34 @@ packages=(
...
+asp export iproute2
+chmod 777 iproute2
+
+cd iproute2
+sed -i 's/'"'"'iptables'"'"' //g' PKGBUILD
+sed -i '/makedepends/d' PKGBUILD
+chmod 666 *
+su nobody -s /bin/bash -c "makepkg -c --skippgpcheck"
+pacman -U "$(su nobody -s /bin/bash -c "makepkg --packagelist")"
```
Comment (wat-ze-hex) [inline travis-ci/vmtest/mkrootfs.sh:112]: ditto, `-r $root` ?

File: travis-ci/vmtest/mkrootfs.sh (hunk 1 above)
Comment (joestringer) [reply travis-ci/vmtest/mkrootfs.sh:111]: Did you hit issues with this?

Without these options, AFAICT `makepkg` just places artifacts into the current directory. With these options, it expects to create the specified directory and insert the artifacts there. I don't see the point in micromanaging all of these settings if the default is sane (and we'll clean up the files in the current directory).

File: travis-ci/vmtest/mkrootfs.sh (hunk 2 above)
Comment (joestringer) [reply travis-ci/vmtest/mkrootfs.sh:103]: Will update.

File: travis-ci/vmtest/mkrootfs.sh (hunk 3 above)
Comment (joestringer) [reply travis-ci/vmtest/mkrootfs.sh:112]: Will update.

File: travis-ci/vmtest/mkrootfs.sh (hunk 1 above)
Comment (joestringer) [reply travis-ci/vmtest/mkrootfs.sh:111]: Hmm, I had no trouble with this on friday but now it keeps complaining. I'll dig a little deeper.

Comment (joestringer) [reply travis-ci/vmtest/mkrootfs.sh]: Looks like it was the original $PWD permissions. I've changed it to move into $root first to create the build dir there, and fixed up related permissions. Should be good now.

**Comment by joestringer** [comment]
@wat-ze-hex not sure off the top of my head, what does `mount` tell you? Is proc already mounted / bind-mounted to somewhere?

**Comment by wat-ze-hex** [comment]
@joestringer 

The issue was in trying to mount `devtmpfs` twice. After #147 it's passing [[0]](https://travis-ci.org/github/wat-ze-hex/libbpf/jobs/673718678):

```
#49 sk_assign:OK
```

The bad news is that I had to blacklist a few other tests due to a crash in `bpf_probe_write_user`, e.g. [[1]](https://travis-ci.org/github/wat-ze-hex/libbpf/jobs/673331993).  

As for the whole bunch of `iproute2` dependencies, turns out that `libelf` also has `systemd`, `dbus`, etc:

```
libelf
├─gcc-libs
│ └─glibc provides glibc>=2.27
│   ├─linux-api-headers provides linux-api-headers>=4.10
│   ├─tzdata
│   └─filesystem
│     └─iana-etc
├─zlib
│ └─glibc
├─bzip2
│ ├─glibc
│ └─bash provides sh
│   ├─readline provides readline>=7.0
│   │ ├─glibc
│   │ ├─ncurses
│   │ │ ├─glibc
│   │ │ └─gcc-libs
│   │ └─ncurses provides libncursesw.so=6-64
│   ├─glibc
│   └─ncurses
├─xz
│ └─bash provides sh
└─curl
  ├─ca-certificates
  │ └─ca-certificates-mozilla
  │   └─ca-certificates-utils provides ca-certificates-utils>=20181109-3
  │     ├─bash
  │     ├─coreutils
  │     │ ├─glibc
  │     │ ├─acl
  │     │ │ └─attr
  │     │ │   └─glibc
  │     │ ├─attr
  │     │ ├─gmp
  │     │ │ ├─gcc-libs
  │     │ │ └─bash provides sh
  │     │ ├─libcap
  │     │ │ ├─glibc
  │     │ │ └─attr
  │     │ └─openssl
  │     │   └─perl
  │     │     ├─gdbm provides gdbm>=1.17
  │     │     │ ├─glibc
  │     │     │ └─bash provides sh
  │     │     ├─db
  │     │     │ ├─gcc-libs
  │     │     │ └─bash provides sh
  │     │     └─glibc
  │     ├─findutils
  │     │ ├─glibc
  │     │ └─bash provides sh
  │     └─p11-kit provides p11-kit>=0.23.19
  │       ├─libp11-kit provides libp11-kit=0.23.20-3
  │       │ ├─glibc
  │       │ ├─libtasn1
  │       │ │ └─glibc
  │       │ └─libffi
  │       │   └─glibc
  │       ├─systemd
  │       │ ├─acl
  │       │ ├─bash
  │       │ ├─cryptsetup
  │       │ │ ├─device-mapper
  │       │ │ │ ├─glibc
  │       │ │ │ └─systemd-libs
  │       │ │ │   ├─glibc
  │       │ │ │   ├─libcap
  │       │ │ │   ├─libgcrypt
  │       │ │ │   │ └─libgpg-error
  │       │ │ │   │   ├─glibc
  │       │ │ │   │   └─bash provides sh
  │       │ │ │   ├─lz4
  │       │ │ │   │ └─glibc
  │       │ │ │   └─xz
  │       │ │ ├─openssl
  │       │ │ ├─popt
  │       │ │ │ └─glibc
  │       │ │ ├─libutil-linux
  │       │ │ ├─json-c
  │       │ │ │ └─glibc
  │       │ │ └─argon2
  │       │ │   └─glibc
  │       │ ├─dbus
  │       │ │ ├─systemd-libs
  │       │ │ ├─expat
  │       │ │ │ └─glibc
  │       │ │ ├─audit
  │       │ │ │ ├─krb5
  │       │ │ │ │ ├─e2fsprogs
  │       │ │ │ │ │ ├─bash provides sh
  │       │ │ │ │ │ └─libutil-linux
  │       │ │ │ │ ├─libldap
  │       │ │ │ │ │ ├─libsasl
  │       │ │ │ │ │ │ └─openssl
  │       │ │ │ │ │ └─e2fsprogs
  │       │ │ │ │ └─keyutils
  │       │ │ │ │   ├─glibc
  │       │ │ │ │   └─bash provides sh
  │       │ │ │ └─libcap-ng
  │       │ │ │   └─glibc
  │       │ │ ├─systemd-libs provides libsystemd.so=0-64
  │       │ │ └─audit provides libaudit.so=1-64
  │       │ ├─iptables
  │       │ │ ├─libnftnl
  │       │ │ │ └─libmnl
  │       │ │ │   └─glibc
  │       │ │ ├─libpcap
  │       │ │ │ ├─glibc
  │       │ │ │ ├─libnl
  │       │ │ │ │ └─glibc
  │       │ │ │ ├─bash provides sh
  │       │ │ │ ├─libusb provides libusbx
  │       │ │ │ │ ├─glibc
  │       │ │ │ │ └─systemd-libs
  │       │ │ │ └─dbus
  │       │ │ ├─libnfnetlink
  │       │ │ │ └─glibc
  │       │ │ ├─libnetfilter_conntrack
  │       │ │ │ ├─libnfnetlink
  │       │ │ │ └─libmnl
  │       │ │ └─bash
  │       │ ├─kbd
  │       │ │ ├─glibc
  │       │ │ └─pam
  │       │ │   ├─glibc
  │       │ │   ├─cracklib
  │       │ │   │ ├─glibc
  │       │ │   │ └─zlib
  │       │ │   ├─libtirpc
  │       │ │   │ └─krb5
  │       │ │   └─pambase
  │       │ ├─kmod
  │       │ │ ├─glibc
  │       │ │ ├─zlib
  │       │ │ ├─openssl
  │       │ │ └─xz
  │       │ ├─hwids
  │       │ ├─libcap
  │       │ ├─libgcrypt
  │       │ ├─systemd-libs
  │       │ ├─libidn2
  │       │ │ └─libunistring
  │       │ │   └─glibc
  │       │ ├─libidn2 provides libidn2.so=0-64
  │       │ ├─lz4
  │       │ ├─pam
  │       │ ├─libelf
  │       │ ├─libseccomp
  │       │ │ └─glibc
  │       │ ├─util-linux
  │       │ │ ├─pam
  │       │ │ ├─shadow
  │       │ │ │ ├─pam
  │       │ │ │ ├─acl
  │       │ │ │ ├─audit
  │       │ │ │ └─audit provides libaudit.so=1-64
  │       │ │ ├─coreutils
  │       │ │ ├─systemd-libs
  │       │ │ ├─libcap-ng
  │       │ │ └─libutil-linux
  │       │ ├─xz
  │       │ ├─pcre2
  │       │ │ ├─gcc-libs
  │       │ │ ├─readline
  │       │ │ ├─zlib
  │       │ │ ├─bzip2
  │       │ │ └─bash
  │       │ ├─audit
  │       │ └─libp11-kit
  │       ├─libp11-kit provides libp11-kit.so=0-64
  │       └─systemd-libs provides libsystemd.so=0-64
  ├─krb5
  ├─libssh2
  │ ├─openssl
  │ └─zlib
  ├─openssl
  ├─zlib
  ├─libpsl
  │ ├─libidn2
  │ ├─libidn2 provides libidn2.so=0-64
  │ └─libunistring
  ├─libnghttp2
  │ └─glibc
  ├─libidn2
  └─libidn2 provides libidn2.so=0-64
```
That wasn't the case a month ago when I built the previous image. I suggest to just leave with it.

Probably the one follow-up that we may do is to fully embrace `systemd` by replacing the current [SysVinit](https://wiki.archlinux.org/index.php/SysVinit) init scripts with `systemd`. Besides it is emphasized that Arch no longer supports SysVinit. In that case kernel selftests will be run as systemd service.
 
[0] https://travis-ci.org/github/wat-ze-hex/libbpf/jobs/673718678
[1] https://travis-ci.org/github/wat-ze-hex/libbpf/jobs/673331993

**Comment by joestringer** [comment]
Awesome! Looks like this custom iproute2 build hack is no longer necessary then.

Thanks, closing.

**Pull request closed without merging**